src/stSearch.js

```diff
@@ -7,12 +7,13 @@ ng.module('smart-table')
         var promise = null;
         var throttle = attr.stDelay || stConfig.search.delay;
         var event = attr.stInputEvent || stConfig.search.inputEvent;
+        var trimSearch = attr.trimSearch || stConfig.search.trimSearch;
 
         attr.$observe('stSearch', function (newValue, oldValue) {
           var input = element[0].value;
           if (newValue !== oldValue && input) {
             ctrl.tableState().search = {};
-            tableCtrl.search(input, newValue);
+            tableCtrl.search(input, newValue, trimSearch);
```

Comment:
That is great, thanks ! However can you actually do the "trim work" on the input rather than passing a flag to the controller API. 
The reason is that I don't want a flag specific to string to be passed to the controller which does not really care whether the consumer uses string, object etc.

Reply:
So it is just about moving the trim work from the controller to the directive ! Great for the tests and everything !